### ISSUE TITLE
Tasks not respecting instance limit

### ISSUE THREAD

**Issue by Novynn** [microsoft/vscode#96959]
Issue Type: <b>Bug</b>

VS Code version: Code 1.44.2 (ff915844119ce9485abfe8aa9076ec76b5300ddd, 2020-04-16T16:36:23.138Z)
OS version: Windows_NT x64 10.0.19041

<b>Repro:</b>
Using the following tasks.json:
```
{
    // See https://go.microsoft.com/fwlink/?LinkId=733558
    // for the documentation about the tasks.json format
    "version": "2.0.0",
    "tasks": [
        {
            "label": "run update.sh",
            "type": "shell",
            "command": "./update.sh",
            "options": {
                "shell": {
                    "executable": "bash.exe",
                    "args": [
                        "-c"
                    ]
                }
            },
            "group": {
                "kind": "build",
                "isDefault": true
            }
        }
    ]
}
```

<b>Observed Behaviour:</b>
Holding down Ctrl+Shift+B (run default build task) will result in multiple tasks running at the same time. This clogs up the "currently running tasks" list especially when a task hangs.

<b>Expected Behaviour:</b>
A notification should tell us that the task is already running. This is what I see using version 1.42.1.

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-6600K CPU @ 3.50GHz (4 x 3504)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
</details>

**Comment by bmartin** [comment]
I've been seeing this too. This causes corrupted builds in my project since I'm in the habit of invoking builds frequently and the build is not safe to running in parallel with itself.

**Comment by alexr00** [comment]
Thank you for reporting this! Duplicate of https://github.com/microsoft/vscode/issues/96999
